### PR TITLE
MP-8458 - Creation of Nox Type "DatabaseGuid"

### PR DESCRIPTION
**Pull request by danield-ap**
- create database configuration
- add to framework test

### CODE REVIEW

File: src/Nox.Types.EntityFramework/Types/DatabaseGuid/DatabaseGuidDatabaseConfigurator.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public void ConfigureEntityProperty(
+        NoxSolutionCodeGeneratorState noxSolutionCodeGeneratorState,
+        EntityTypeBuilder builder,
+        NoxSimpleTypeDefinition property,
+        Entity entity,
+        bool isKey)
+    {
+        builder
+            .Property(property.Name)
+            .IsRequired(isKey || property.IsRequired)
```
[the remark this answers is not in the document]
Comment (danield-ap) [reply src/Nox.Types.EntityFramework/Types/DatabaseGuid/DatabaseGuidDatabaseConfigurator.cs]: `isKey` removed.

File: tests/Nox.Tests/DatabaseIntegrationTests/SqliteIntegrationTests.cs [hunk 2]
```diff
@@ -174,6 +174,7 @@ Plain text
             WeightTestField = Weight.From(weight, new WeightTypeOptions() { Units = WeightTypeUnit.Pound, PersistAs = persistWeightUnitAs }),
             DistanceTestField = Distance.From(distance, new DistanceTypeOptions() { Units = DistanceTypeUnit.Mile, PersistAs = persistDistanceUnitAs }),
             DatabaseNumberTestField = DatabaseNumber.FromDatabase(databaseNumber), //SQLite supports AutoIncrement only for column of type INTEGER PRIMARY KEY  https://www.sqlite.org/autoinc.html
+            DatabaseGuidTestField = DatabaseGuid.FromDatabase(databaseGuid),
```
[the remark this answers is not in the document]
Comment (danield-ap) [reply tests/Nox.Tests/DatabaseIntegrationTests/SqliteIntegrationTests.cs:177]: Same as Integer it doesn't have an option for generating Guid in Sqlite. 
And also it needs a specific configuration to generate for Postgresql, that I added to `PostgresDatabaseGuidDatabaseConfiguration.cs`.

File: src/Nox.Types.EntityFramework/Types/DatabaseGuid/DatabaseGuidDatabaseConfigurator.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    /// <summary>
+    /// Gets the name of the key property for the DatabaseGuid type.
+    /// This method is called by the NoxSolutionCodeGeneratorState to retrieve the name of the key property for the type.
+    /// </summary>
+    /// <param name="key">The NoxSimpleTypeDefinition representing the key property.</param>
+    /// <returns>The name of the key property.</returns>
+    public string GetKeyPropertyName(NoxSimpleTypeDefinition key) => key.Name;
+
+    public virtual string? DefaultValue()
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Types.EntityFramework/Types/DatabaseGuid/DatabaseGuidDatabaseConfigurator.cs:55]: Where do you use the Default value?
Should it be used in the "HasDefaultValue" if not the DefaultValue() is not null?

**Review by gjorgji-grgovski-authoritypartners — CHANGES_REQUESTED** [review]

File: src/Nox.Types.EntityFramework/Types/DatabaseGuid/DatabaseGuidDatabaseConfigurator.cs (hunk 3 above)
Comment (danield-ap) [reply src/Nox.Types.EntityFramework/Types/DatabaseGuid/DatabaseGuidDatabaseConfigurator.cs]: Yes, that's the reason I created it, but I forgot to include that part.
It's included now.

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Pull request merged**